### PR TITLE
ROC-5811: use next working day for deadlines where appropriate

### PR DESCRIPTION
**Pull request by KiranVarma02**
**Before creating a pull request make sure that:**

- [x] commit messages are meaningful and follow good commit message guidelines
- [x] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ROC-5811

### Change description ###
Get mediation deadline day calculated within the claim and send it for frontend

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[* ] No
```

### CODE REVIEW

File: domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/Claim.java [hunk 1]
```diff
@@ -225,4 +225,11 @@ public String toString() {
         return ReflectionToStringBuilder.toString(this, ourStyle());
     }
 
+    @JsonProperty("mediationDeadline")
+    public Optional<LocalDate> getMediationDeadline() {
+        if (this.respondedAt != null) {
+            return Optional.ofNullable(new NextWorkingDay(this.respondedAt.toLocalDate().plusDays(5)).getDate());
+        }
+        return Optional.empty();
+    }
```
Comment (andrewwa-kainos) [inline domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/Claim.java]: Might be a bit clearer what's going on:
```
return Optional.ofNullable(this.respondedAt)
    .map(LocalDateTime::toLocalDate)
    .map(date -> date.plusDays(5))
    .map(NextWorkingDay::new)
    .map(NextWorkingDay::getDate);
```

Comment (andrewwa-kainos) [inline domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/Claim.java]: As discussed, please add unit tests

Comment (KiranVarma02) [reply domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/Claim.java]: done

Comment (KiranVarma02) [reply domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/Claim.java]: done

File: domain-model/src/test/java/uk/gov/hmcts/cmc/domain/models/ClaimTest.java [hunk 2]
```diff
@@ -97,6 +97,26 @@ public void shouldHaveNoValidationMessageWhenInstanceIsValidPast() {
...
     }
 
+    @Test
+    public void shouldGetMediationDeadLineDate() {
+        LocalDateTime someDate = LocalDate.of(2019,7, 7).atStartOfDay();
+        Claim claim = SampleClaim.builder()
+            .withRespondedAt(someDate)
+            .build();
+        claim.getMediationDeadline().ifPresent(
+            result -> assertThat(result).isEqualTo(LocalDate.of(2019,7, 12)));
```
Comment (andrewwa-kainos) [inline domain-model/src/test/java/uk/gov/hmcts/cmc/domain/models/ClaimTest.java]: I think in these cases we want the tests to fail if the deadline is not present

File: domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/Claim.java [hunk 3]
```diff
@@ -225,4 +225,21 @@ public String toString() {
         return ReflectionToStringBuilder.toString(this, ourStyle());
     }
 
+    @JsonProperty("mediationDeadline")
+    public Optional<LocalDate> getMediationDeadline() {
+        return Optional.ofNullable(this.respondedAt)
+            .map(LocalDateTime::toLocalDate)
+            .map(date -> date.plusDays(5))
+            .map(NextWorkingDay::new)
```
Comment (markry-kainos) [inline domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/Claim.java]: why are we mapping to NextWorkingDay 🤔 - this doesn't actually apply any next working day logic does it?

Comment (KiranVarma02) [reply domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/Claim.java]: this is non java 8 way of writing it - `            return Optional.ofNullable(new NextWorkingDay(this.respondedAt.toLocalDate().plusDays(5)).getDate());
`

Not sure how i can explain, i will try - getDate uses a class variable which is set in NextWorkingDay. So, i think i have to do it like the way it is..

**Pull request closed without merging**